analyses/org.eclipse.tracecompass.incubator.dpdk.core/src/org/eclipse/tracecompass/incubator/internal/dpdk/core/ethdev/throughput/analysis/AbstractDpdkEthdevThroughputDataProvider.java

```diff
@@ -0,0 +1,249 @@
+
+                int rxQsQuark = ss.optQuarkRelative(nicQuark, DpdkEthdevThroughputAttributes.RX_Q);
+                int txQsQuark = ss.optQuarkRelative(nicQuark, DpdkEthdevThroughputAttributes.TX_Q);
+
+                nodes.addAll(getQueuesTree(ss, rxQsQuark, nicName, nicId));
+                nodes.addAll(getQueuesTree(ss, txQsQuark, nicName, nicId));
+            }
+        } catch (AttributeNotFoundException e) {
+            Activator.getInstance().logError("Error getting the root attribute of " + DpdkEthdevThroughputAttributes.NICS); //$NON-NLS-1$
+        }
+        return new TmfTreeModel<>(Collections.emptyList(), nodes);
+    }
+
+    /**
+     * Get parts of the XY tree that are related to the RX and TX queues
+     *
+     * @param ss
+     *            The state system
+     * @param qsQuark
+     *            The quark of the queues list
+     * @param nicName
+     *            The name of the NIC to which the queues are attached
+     * @param nicId
+     *            The id of the related port
+     * @return a list of {@link TmfTreeDataModel}
+     */
+    protected List<TmfTreeDataModel> getQueuesTree(ITmfStateSystem ss, int qsQuark, String nicName, long nicId) {
+        int i = 0;
+        List<TmfTreeDataModel> nodes = new ArrayList<>();
+        boolean isRxQueue = true;
+
+        try {
+            if (DpdkEthdevThroughputAttributes.TX_Q.equals(ss.getAttributeName(qsQuark))) {
+                isRxQueue = false;
+            }
+
+            long qsId = getId(qsQuark);
+            nodes.add(new TmfTreeDataModel(qsId, nicId, Collections.singletonList(isRxQueue ? RX_LABEL : TX_LABEL), false, null));
+
+            for (Integer queueQuark : ss.getSubAttributes(qsQuark, false)) {
+                String queueName = ss.getAttributeName(queueQuark);
+                long queueId = getId(queueQuark);
+
+                // Get color and style for the queue
+                Pair<String, String> colorPair = COLOR_LIST.get(i % COLOR_LIST.size());
+                String seriesStyle = SUPPORTED_STYLES.get((i / COLOR_LIST.size()) % SUPPORTED_STYLES.size());
+
+                nodes.add(new TmfTreeDataModel(queueId, qsId, Collections.singletonList(queueName), true,
+                        new OutputElementStyle(BASE_STYLE, ImmutableMap.of(
+                                StyleProperties.COLOR, isRxQueue ? colorPair.getFirst() : colorPair.getSecond(),
+                                StyleProperties.SERIES_STYLE, seriesStyle,
+                                StyleProperties.STYLE_NAME, nicName + '/' + (isRxQueue ? RX_LABEL : TX_LABEL) + '/' + queueName))));
+
+                i++;
+            }
+        } catch (IndexOutOfBoundsException e) {
+            Activator.getInstance().logWarning("A DPDK event (" + DpdkEthdevEventLayout.eventEthdevConfigure() + ") is missing", e); //$NON-NLS-1$ //$NON-NLS-2$
+        }
+        return nodes;
+    }
+
+    @Override
```

Comment:
Do we still need this?

Reply:
@MatthewKhouzam: Do you mean the AbstractDpdkEthdevThroughputDataProvider class?